### DISCUSSION TITLE
Best frigate-settings in low light and difficult environment (a possible way to find out)

### DISCUSSION THREAD

**Discussion by KonTiki1957** [blakeblackshear/frigate#966] [Ask A Question]
Hi to everybody! Well I'm struggling to find the best frigate settings mainly during the night (suggestions are welcome) and in low light conditions. My HW: raspi4, docker, 4 dahua cameras, for the time being no Coral (hopefully it will arrive soon) and no extra IR lights which I may decide to install later. Now since so far as I understand frigate works in a sort of two step mode: first it scans for movements in the pictures and then "if" there is movement it starts to try to understand if the movement is produced by any objects it is required to identify. As I said I'm trying to understand where it fails hence it would be interesting to get a notification any time it reveals a movement (with pictures) and not only "after" the second step, in order to be able to discriminate which is the phase it is failing and trim the settings accordingly. But I have no idea how to accomplish this (I mean to get the first phase-notification) and I would be very grateful to anybody helping me in the endeavor.

**Comment by hawkeye217** [comment]
Depending upon your camera resolution, it may help to tweak the ``motion`` parameters here: 

https://blakeblackshear.github.io/frigate/configuration/advanced

I have a similar setup with a Pi4 and am still trying to find a good balance for detecting objects day and night. With a motion threshold set too low, Frigate unnecessarily uses a lot of CPU looking for objects because of moving leaves or shadows. But with a threshold set too high to avoid that issue during the day, motion is not detected at night when a person walks in the field of view of the camera.  It would be helpful to be able to set the motion threshold settings via MQTT or something...

**Reply by KonTiki1957** [reply]
Thank you for replying. That's the page where I started from. I have a very complicated environment, 3 cameras facing each the cardinal directions E, S, W and one in a semi closed porch; many plants in the garden and even an olive tree among them (you can imagine what the shadows do during wind days...). Anyway for the time being, sharing your concept about the necessity of having different configurations meeting the different needs at different day hours, I have come up with the following solution: I have two different frigate configurations (day and night involving not only the thresholds but other items and even masks) and three different configurations for the four dahua cameras. At given times through cronjobs governed by the host os the configurations get changed and in the case of the frigate configurations the addon gets restarted to activate them. Now I'm only facing the problem that randomly the addon gets stuck in the process and I have to "rebuild" it through the button in the addon to get it going again. I activated now the "Watchdog" function of the addon to see if this stops beeing an issue. Anyway in this way the system seems to be reliable, at least during the day hours. I will test it on a longer period during night operation even if I think I will have to install at least two IR illuminators. Btw I'm still waiting for the Coral in the hope that the increased computing speed will offer a better reliability as well especially when data from more than one camera have to be processed.

**Reply by hawkeye217** [reply]
I definitely understand your problem. I've got some pretty wild issues on windy days too.  

Having multiple configurations is a good workaround.  It might be worthwhile to have an automation that just restarts the Frigate container at sunset with the "night" configuration and then again at sunrise with the "day" configuration.  I'll play around with that.

The Coral is a great little device and will definitely reduce CPU usage and improve detection speed for you.
